src/main/java/de/zalando/aruha/nakadi/controller/EventPublishingController.java

```diff
@@ -50,39 +53,55 @@ public EventPublishingController(final TopicRepository topicRepository,
     @Timed(name = "post_events", absolute = true)
     @RequestMapping(value = "/event-types/{eventTypeName}/events", method = POST)
     public ResponseEntity postEvent(@PathVariable final String eventTypeName, @RequestBody final String event,
-            final NativeWebRequest nativeWebRequest) {
+                                    final NativeWebRequest nativeWebRequest) {
         LOG.trace("Received event {} for event type {}", event, eventTypeName);
 
         try {
             final EventType eventType = eventTypeRepository.findByName(eventTypeName);
 
-            final Optional<ValidationError> error = validateSchema(event, eventType);
+            final JSONObject eventAsJson = validateSchema(event, eventType);
+
+            String partitionId = applyPartitioningStrategy(eventType, eventAsJson);
+
+            topicRepository.postEvent(eventTypeName, partitionId, event);
+            return status(HttpStatus.CREATED).build();
 
-            if (error.isPresent()) {
-                final Problem p = Problem.valueOf(MoreStatus.UNPROCESSABLE_ENTITY, error.get().getMessage());
-                return create(p, nativeWebRequest);
-            } else {
-                // Will be replaced later:
-                final String partitionId = "1";
-                topicRepository.postEvent(eventTypeName, partitionId, event);
-                return status(HttpStatus.CREATED).build();
-            }
         } catch (NoSuchEventTypeException e) {
             LOG.debug("Could not process event.", e);
             return create(e.asProblem(), nativeWebRequest);
+        } catch (final EventValidationException e) {
+            LOG.debug("Event validation error: {}", e.getValidationError().getMessage());
+            return create(e.asProblem(), nativeWebRequest);
         } catch (final NakadiException e) {
             LOG.error("error posting to partition", e);
             return create(e.asProblem(), nativeWebRequest);
         }
     }
 
-    private Optional<ValidationError> validateSchema(final String event, final EventType eventType) {
+    private String applyPartitioningStrategy(final EventType eventType, final JSONObject eventAsJson) throws InvalidOrderingKeyFieldsException {
+        String partitionId;
+        if (!eventType.getOrderingKeyFields().isEmpty()) {
+            partitionId = orderingKeyFieldsPartitioningStrategy.calculatePartition(eventType, eventAsJson, 8);
```

Comment:
I wanted to avoid the costly operation of asking Kafka for the number of partitions every time. Hence, I asked whether we expect the number of partitions for a topic to change. But a hardcoded `8` is not good, I agree. I will ask Kafka and cache the number of partitions per event type.